src/crimson/osd/osd_operations/client_request.cc

```diff
@@ -354,7 +354,8 @@ ClientRequest::process_op(
     }
 
     std::set<snapid_t> snaps = snaps_need_to_recover();
-    if (!snaps.empty()) {
+    if (!snaps.empty() &&
+        pg->is_missing_head_and_clones(m->get_hobj().get_head())) {
```

Comment:
@cyx1231st, seems like this might positively affect client req performance.

Reply:
We do still take obc read locks on both the head and clone later on, so I think the first line should be

"osd/../_client_request: skip recover_missing_snaps if none are missing"

Reply:
Originally I've tried to emphasize that fact that we could avoid an *additional* obc read lock.
Renamed as suggested, thanks!